File: PusherPlatform/src/main/java/com/pusher/platform/logger/Logger.java
package com.pusher.platform.logger;

/**
 * You have 3 guesses to figure out what this does.
 * */
public interface Logger {
    void log(String message);
    void log(Throwable e, String message);
    void log(String message, Object... arguments);
    void log(Throwable e, String message, Object... arguments);
    void log(Object object);
}

File: library/src/main/kotlin/com/pusher/platform/logger/Logger.kt
package com.pusher.platform.logger

interface Logger {

    fun verbose(message: String, error: Error?)
    fun debug(message: String, error: Error?)
    fun info(message: String, error: Error?)
    fun warn(message: String, error: Error?)
    fun error(message: String, error: Error?)
}





